compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnvilModuleDescriptor.kt

```diff
@@ -23,47 +22,50 @@ public interface AnvilModuleDescriptor : ModuleDescriptor {
     lookupLocation: LookupLocation = FROM_BACKEND
   ): ClassDescriptor?
 
-  public fun getClassesAndInnerClasses(ktFile: KtFile): List<KtClassOrObject>
-
-  public fun getKtClassOrObjectOrNull(fqName: FqName): KtClassOrObject?
+  public fun getClassAndInnerClassReferences(ktFile: KtFile): List<Psi>
 
   public fun getClassReference(clazz: KtClassOrObject): Psi
 
   public fun getClassReference(descriptor: ClassDescriptor): Descriptor
 
-  public fun getClassReferenceOrNull(fqName: FqName): ClassReference?
+  /**
+   * Attempts to resolve the [FqName] to a [ClassDescriptor] first, then falls back to a
+   * [KtClassOrObject] if the descriptor resolution fails. This will happen if the code being
+   * parsed was generated as part of the compilation round for this module.
+   *
+   * If [preferDescriptor] is `false`, then the order of resolution is reverted.
+   */
+  public fun getClassReferenceOrNull(
+    fqName: FqName,
+    preferDescriptor: Boolean = true
+  ): ClassReference?
```

Comment:
Could you include why we prefer descriptors over PSI in the comment?

Reply:
I honestly don't know why @RBusarow decided for this approach. Maybe he can clarify?